src/rgw/rgw_sync_module_es.cc

```diff
@@ -564,18 +565,33 @@ class RGWElasticInitConfigCBCR : public RGWCoroutine {
   int operate() override {
     reenter(this) {
       ldout(sync_env->cct, 0) << ": init elasticsearch config zone=" << sync_env->source_zone << dendl;
+      yield call(new RGWReadRESTResourceCR<ESInfo> (sync_env->cct,
+                                                    conf->conn.get(),
+                                                    sync_env->http_manager,
+                                                    "/", nullptr, &es_info));
+      if (retcode < 0) {
+        return set_cr_error(retcode);
+      }
+
       yield {
         string path = conf->get_index_path();
+        ldout(sync_env->cct, 5) << "got elastic version=" << es_info.version.to_str() << dendl;
 
         es_index_settings settings(conf->num_replicas, conf->num_shards);
         es_index_mappings mappings;
+        if (es_info.version >= ESVersion(5,0)) {
+          ldout(sync_env->cct, 0) << "elasticsearch: using text type for string index mappings " << dendl;
+          mappings.string_type = ESType::Text;
```

Comment:
@theanalyst note migration instructions here:
https://www.elastic.co/blog/strings-are-dead-long-live-strings
`string` converts into either `text` or `keyword`, depending on the `analyzed` state. We also need to make sure the `index` field is being created correctly (as code is written now, it doesn't create it).

Reply:
Thanks for the find, I'll modify dump_type to handle these cases

Reply:
I'm seeing the same behaviour even without the patch in ES 5.6, we get the double mapping to both text and keyword. I assume name fields should be text types and most of the fields like IDs should be keywords

Reply:
So reading more into this it seems even mapping types are already deprecated and going away in the next ES version. https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html, not sure how exactly we're supposed to be mapping types then. For now I'll decide based on "analyzed" field whether we dump a keyword or a text if the type set is not a string already.